Sources/Auth/Services/App/AuthRequestService.swift

```diff
@@ -1,8 +1,30 @@
 import Foundation
+import WalletConnectUtils
+import WalletConnectKMS
 
 actor AuthRequestService {
+    private let networkingInteractor: NetworkInteracting
+    private let appMetadata: AppMetadata
+    private let kms: KeyManagementService
 
-    func request(params: RequestParams, topic: String) async throws {
+    init(networkingInteractor: NetworkInteracting,
+         kms: KeyManagementService,
+         appMetadata: AppMetadata) {
+        self.networkingInteractor = networkingInteractor
+        self.kms = kms
+        self.appMetadata = appMetadata
+    }
 
+    func request(params: RequestParams, topic: String) async throws {
+        let pubKey = try kms.createX25519KeyPair()
+        let responseTopic = pubKey.rawRepresentation.sha256().toHexString()
+        let requester = AuthRequestParams.Requester(publicKey: pubKey.hexRepresentation, metadata: appMetadata)
+        let issueAt = ISO8601DateFormatter().string(from: Date())
+        let payload = AuthPayload(requestParams: params, iat: issueAt)
+        let params = AuthRequestParams(requester: requester, payloadParams: payload)
+        let request = JSONRPCRequest<AuthRequestParams>(method: "wc_authRequest", params: params)
```

Comment:
What do you think of using `RPCRequest` in here? Seems like a good place to use it

Reply:
I was waiting for your first demo integration 😄  but ok, let's give it a shot.